tests/test_rpc.py

```
@@ -73,6 +101,190 @@ def test_listcoins(lianad, bitcoind):
+    wait_for(lambda: len(lianad.rpc.listcoins()["coins"]) == 4)
+    res = lianad.rpc.listcoins(["unconfirmed"], [])["coins"]
+    outpoint_d = res[0]["outpoint"]
+
+    # We now have all four statuses.
+    assert (
+        len(lianad.rpc.listcoins(["unconfirmed"])["coins"])
+        == len(lianad.rpc.listcoins(["confirmed"])["coins"])
+        == len(lianad.rpc.listcoins(["spending"])["coins"])
+        == len(lianad.rpc.listcoins(["spent"])["coins"])
+        == 1
+    )
+    assert (
+        len(lianad.rpc.listcoins()["coins"])
+        == len(
+            lianad.rpc.listcoins([], [outpoint_a, outpoint_b, outpoint_c, outpoint_d])[
+                "coins"
+            ]
+        )
+        == len(
+            lianad.rpc.listcoins(["unconfirmed", "confirmed", "spending", "spent"])[
+                "coins"
+            ]
+        )
+        == len(
+            lianad.rpc.listcoins(
+                ["unconfirmed", "confirmed", "spending", "spent"],
+                [outpoint_a, outpoint_b, outpoint_c, outpoint_d],
+            )["coins"]
+        )
+        == 4
+    )
+
+    # We can filter for specific statuses/outpoints.
+    assert (
+        sorted(
+            lianad.rpc.listcoins(["spending", "spent"])["coins"],
+            key=lambda c: c["outpoint"],
+        )
+        == sorted(
+            lianad.rpc.listcoins(["spending", "spent"], [outpoint_a, outpoint_c])[
+                "coins"
+            ],
+            key=lambda c: c["outpoint"],
+        )
+        == sorted(
+            lianad.rpc.listcoins(
+                ["unconfirmed", "confirmed", "spending", "spent"],
+                [outpoint_a, outpoint_c],
+            )["coins"],
+            key=lambda c: c["outpoint"],
+        )
+        == sorted(
+            lianad.rpc.listcoins(
+                ["spending", "spent"], [outpoint_a, outpoint_b, outpoint_c, outpoint_d]
+            )["coins"],
+            key=lambda c: c["outpoint"],
+        )
+    )
```

Great functional test too. A possible good addition is to sanity check the errors returned in case of invalid inputs. For instance make sure we catch an invalid status or an invalid outpoint. This would have caught the fact we actually ignore the error on malformed outpoints as i pointed above.

Example of such a sanity check:
https://github.com/wizardsardine/liana/blob/38d0fafac2b3cbdba57070173dc0136e20fd7019/tests/test_rpc.py#L610-L612

---

That's a good idea thanks. I've added some error checks to the test.